pkg/apis/discovery/v1alpha1/defaults.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 var (
-	defaultAddressType = discoveryv1alpha1.AddressTypeIP
+	defaultAddressType = discoveryv1alpha1.AddressTypeIPv4
```

Comment:
Why do we believe that we need a default?

Reply:
hmm, here is a thought. 

What we really care about is not to have mixed family/types in the slice, right? We don't need the user to tell us that this is an ipv4 or ipv6. We can set that.  In other words, this field can be `readonly`.  And in this case, it does not need to be defaulted or anything.

Reply:
Unfortunately we can't do this automatically even if we wanted to since address type is immutable and an EndpointSlice with no addresses is valid. Additionally, it's likely that there are addresses that would be valid for multiple address types. I think having a default here is reasonable, and won't take much or any additional effort from EndpointSlice producers.

Reply:
Does AddressType need to be immutable?  If we're adding proper IP validation - we can reject any change that is inconsistent.

E.g., if a user changes all of the addresses from v4 to v6 and changes the addresstype at the same time, is that invalid?  It's not particularly different than deleting one and adding another - does the fact that it has the same metadata.name represent a problem?

I don't see why the default here is needed?  If it is immutable, the creator has to decide up front.  Choosing not to decide, with a default, means you have chosen the default.

Reply:
I think keeping AddressType immutable will help simplify logic for EndpointSlice consumers. As an example, kube-proxy watches for EndpointSlice creates, updates, and deletes, and will only process an EndpointSlice if it matches an AddressType it can handle. If we allowed AddressType to change, it could mean consumers would need to add special handling on updates to see if the AddressType had changed. If it was changing from an AddressType that it did support to one it didn't, it would then have to treat the action more like a delete. 

It's worth noting that Service IPFamily is immutable, so this will not be an issue for the primary use case here. I can't think of many use cases where it would make sense to change an AddressType, and in those cases I think it's reasonable to expect EndpointSlices to be replaced (create+delete) instead of updated. That's at least how I think most consumers would prefer to handle the events.

As far as a default, I don't have a strong opinion here, but would lean towards keeping something. It doesn't cost us anything and is closer to backwards compatibility than not having a default value. We have a default for protocol (TCP) as well as port name (""), neither of which are necessary, but could be helpful in requiring a bit less yaml/config to use EndpointSlices. Defaults don't prevent consumers from being able to specify their own values, they just help simplify common configs.

Reply:
Since key 1.17 PRs are dependent on this PR and code freeze is all too soon, I've gone ahead and removed the default value for address type here.

Reply:
Can remove this variable now